### PR TITLE
Fix WinForms BlazorWebView dispose logic

### LINKED ISSUE #1242: NullReference Exception during exit from 'Microsoft.AspNetCore.Components.WebView.WebView2'
### Describe the bug
Closing a Form with embedded WebView will throw a NullReference Exception.

### To Reproduce
Clone and run the demo repository for this bug:
[JSTearDownCrash](https://github.com/b-straub/JSTearDownCrash)

### Exceptions (if any)
`
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Microsoft.AspNetCore.Components.WebView.WebView2
  StackTrace:
   at Microsoft.AspNetCore.Components.WebView.WebView2.WebView2WebViewManager.SendMessage(String message) in /_/src/Components/WebView/Platforms/WebView2/src/WebView2WebViewManager.cs:line 56
`

### Further technical details

When JS will be called during disposing a Razor component a message will be send through the 'WebView2WebViewManager'.
`protected override void SendMessage(string message)
            => _webview.CoreWebView2.PostWebMessageAsString(message);
`
At this point `_webview.CoreWebView2` has been already disposed. It may be questionable whether js events are some kind of an unmanaged resource, but the shutdown should handle this case gracefully.

An example for this behavior can be found here [ResizeListenerService.Dispose](https://github.com/Garderoben/MudBlazor/blob/61fea2063a8da7a14258a6b111aee1d3656bb501/src/MudBlazor/Services/ResizeListener/ResizeListenerService.cs#L198)

The following code change would do this:
`
protected override void SendMessage(string message)
            => _webview.CoreWebView2?.PostWebMessageAsString(message);
`
However, this might be a too simple solution.

### PR DESCRIPTION
**Pull request by Eilon**
The fix is quite trivial: Make sure we dispose things in the right order!

Fixes #1242

### CODE REVIEW

**Comment by Eilon** [comment]
The test failures are all unrelated Android failures that are known, and also fixed by now.

File: src/BlazorWebView/src/WindowsForms/BlazorWebView.cs [hunk 1]
```diff
@@ -175,17 +175,17 @@ private void HandleRootComponentsCollectionChanged(object sender, NotifyCollecti
...
+				_webviewManager?.Dispose();
             }
-        }
+			base.Dispose(disposing);
+		}
 
-        /// <inheritdoc />
-        protected override ControlCollection CreateControlsInstance()
+		/// <inheritdoc />
+		protected override ControlCollection CreateControlsInstance()
```
Comment (TanayParikh) [inline src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:188]:
```suggestion
        /// <inheritdoc />
        protected override ControlCollection CreateControlsInstance()
```

Looks like spaces got converted to tabs.

**Review by javiercn — APPROVED** [review]

File: src/BlazorWebView/src/WindowsForms/BlazorWebView.cs (hunk 1 above)
Comment (Eilon) [reply src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:188]: A bot will clean this up later :)

File: src/BlazorWebView/src/WindowsForms/BlazorWebView.cs [hunk 2]
```diff
@@ -175,17 +175,17 @@ private void HandleRootComponentsCollectionChanged(object sender, NotifyCollecti
...
         protected override void Dispose(bool disposing)
         {
-            base.Dispose(disposing);
-
             if (disposing)
             {
-                _webviewManager?.Dispose();
-                _webview?.Dispose();
+				// Dispose this component's contents before calling base.Dispose() because that will dispose the WebView2 control, likely
+				// preventing user-written disposal logic from working (because it might try to use Blazor stuff, which wouldn't work anymore).
```
Comment (SteveSandersonMS) [inline src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:181]: The comment doesn't explain why it's no longer necessary to call `_webview?.Dispose()`. Could you clarify?

Comment (Eilon) [reply src/BlazorWebView/src/WindowsForms/BlazorWebView.cs:181]: It does sort of say it. It says that base.Dispose() will dispose it, so we don't need to. (The problem was two-fold: 1. We were doing unnecessary work; 2. We did the unnecessary work in a broken way.)

**Pull request merged**